core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -332,31 +334,79 @@ default ExecutorService makeContextAware(ExecutorService executor) {
      * Registers {@code callback} to be run when re-entering this {@link RequestContext}, usually when using
      * the {@link #makeContextAware} family of methods. Any thread-local state associated with this context
      * should be restored by this callback.
+     *
+     * @param callback a {@link Consumer} whose argument is this context
+     */
+    void onEnter(Consumer<? super RequestContext> callback);
+
+    /**
+     * Registers {@code callback} to be run when re-entering this {@link RequestContext}, usually when using
+     * the {@link #makeContextAware} family of methods. Any thread-local state associated with this context
+     * should be restored by this callback.
      */
-    void onEnter(Runnable callback);
+    default void onEnter(Runnable callback) {
```

Comment:
I tried to remove the Runnable versions but it seems to make the code that does not require RequestContext way more complex. A user sometimes even has to convert a method reference to a lambda expression:

    ctx.onEnter(span::close);
    // vs.
    ctx.onEnter(myCtx -> span.close());